spec/controllers/category_controller_spec.rb

```diff
@@ -0,0 +1,233 @@
+require "rails_helper"
+include SessionsHelper
+
+RSpec.describe CategoriesController, type: :controller do
+  let(:category) {FactoryBot.create :category}
+  let(:room) {FactoryBot.create :room, category_id: category.id}
+  subject {FactoryBot.create :category}
+  let(:valid_attributes) {FactoryBot.attributes_for :category}
+  let(:invalid_attributes) {FactoryBot.attributes_for :category, name: ""}
+  
+  context "not login" do
+    describe "GET #index" do
+      let!(:categories) { FactoryBot.create_list :category, 3 }
+
+      it "assigns @categories" do
+        expect(assigns(:categories)).to eq assigns(:categories)
+      end
+
+      it "not login index" do
+        get :index
+        expect(response).to redirect_to login_url
+      end
+    end
+
+    describe "GET #new" do
+      it "assigns @category" do
+        expect(assigns(:category)).to eq assigns(:category)
+      end
+
+      it "not login new" do
+        get :index
+        expect(response).to redirect_to login_url
+      end
+    end
+
+    describe "GET #edit" do
+      it "assigns @category" do
+        expect(assigns(:category)).to eq assigns(:category)
```

Comment:
- gì thế em expect thế này thì lúc nào nó chả đúng
- expect(assigns(:category)).to eq [..] 
cho các giá trị của category em expect select ra được vào